components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1042,11 +1071,6 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		}
 
 		// we're not yet finalizing - start the process
-		snc, err := m.manager.connectToWorkspaceDaemon(ctx, *wso)
```

Comment:
@kylos101 this is the source of the error. The connection to ws-daemon should not be part of the retries. The change connects to ws-daemon before and limits the retries to five minutes.